Lib/test/test_pdb.py

```diff
@@ -1798,6 +1798,29 @@ def test_pdb_issue_gh_101517():
     (Pdb) continue
     """
 
+def test_pdb_issue_gh_104301():
+    """See GH-104301
+
+    Make sure that ambiguous statements prefixed by '!' are properly disambiguated
+
+    >>> with PdbTestInput([
+    ...     '! n = 42',  # disambiguated statement: reassign the name n
+    ...     'n',         # advance the debugger into the print()
+    ...     'continue'
+    ... ]):
+    ...     n = -1
+    ...     import pdb; pdb.Pdb(nosigint=True, readrc=False).set_trace()
+    ...     print(n)
+    > <doctest test.test_pdb.test_pdb_issue_gh_104301[0]>(8)<module>()
+    -> print(n)
+    (Pdb) ! n = 42
+    (Pdb) n
+    42
```

Comment:
I'd print "n = 42", so that it's clear from reading the test that this 42 is not printed because of the "n" command on line 1808.